sdk/src/main/scala/com/horizen/SidechainMemoryPool.scala

```diff
@@ -64,9 +70,20 @@ class SidechainMemoryPool(unconfirmed: TrieMap[String, SidechainMemoryPoolEntry]
     filter(t => !txs.exists(_.id == t.id))
   }
 
+  /**
+   * Filters the txs cointained in the mempool, recalculates the used bytes, and return an instance of itself.
+   * @param condition to use to filter the transactions
+   */
   override def filter(condition: SidechainTypes#SCBT => Boolean): SidechainMemoryPool = {
+    usedPoolSizeBytes = 0
     unconfirmed.retain { (k, v) =>
```

Comment:
I'm just thinking if it is correct that we use `retain` function here?
I guess it must be `filter` and then return `new SidechainMemoryPool(res)`
otherwise, anyone from outside may wrongly remove some txs.